src/librustc/hir/def_id.rs

```diff
@@ -139,8 +139,21 @@ impl DefId {
     }
 
     #[inline]
-    pub fn to_local(self) -> LocalDefId {
-        LocalDefId::from_def_id(self)
+    pub fn as_local(self) -> Option<LocalDefId> {
+        if self.is_local() {
```

Comment:
Could use https://doc.rust-lang.org/nightly/std/primitive.bool.html#method.then here.

Reply:
I prefer to only use combinators for containers, when the combinator better expresses the intent of the operation wrt container elements.

The other reason I wouldn't use `bool::then` is because the value of the "then" case is created *unconditionally*, and then discarded by the combinator.

Had `bool::then` instead been, say, `&&`, the short-circuit semantics of that would be somewhat more apparent. But the only thing like that, which works today, AFAIK, is:
```rust
self.is_local()
    && return Some(LocalDefId { index: self.index });
None
```